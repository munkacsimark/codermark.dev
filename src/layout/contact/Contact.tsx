import React from 'react';
import { viewNames } from '../../view-router';
import styles from './Contact.module.scss';

interface IFormData {
  name: string,
  email: string,
  message: string,
}

const Contact: React.FunctionComponent = () => {

  const nameInput = React.createRef<HTMLInputElement>();
  const emailInput = React.createRef<HTMLInputElement>();
  const messageInput = React.createRef<HTMLTextAreaElement>();

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    /*event.preventDefault();
    const data: IFormData = {
      name: nameInput.current ? nameInput.current.value : '',
      email: emailInput.current ? emailInput.current.value : '',
      message: messageInput.current ? messageInput.current.value : '',
    }
    const fetchOptions: RequestInit = {
      method: 'POST',
      headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
      body: JSON.stringify(data)
    };
    fetch('https://formspree.io/munkacsimark@gmail.com', fetchOptions)
      .catch((error: Error) => console.error(error));*/
  }

  return (
    <section
      id={viewNames.CONTACT}
      className={styles.section}>
      <h2 className={styles.title}>
        Contact
      </h2>
      <div className={styles.text}>
        Want to get in touch or talk about a project? Feel free to contact me here.
      </div>
      <form
        action="https://formspree.io/munkacsimark@gmail.com"
        method="POST"
        className={styles.form}
        onSubmit={handleSubmit}>
        <input
          ref={nameInput}
          className={styles.input}
          type="text"
          name="name"
          placeholder="What's your name?"/>
        <input
          ref={emailInput}
          className={styles.input}
          type="email"
          name="email"
          placeholder="What's your email?"/>
        <textarea
          ref={messageInput}
          className={styles.textarea}
          placeholder="How can I help you?"></textarea>
        <button
          className={styles.sendButton}
          type="submit">
          Send message
        </button>
      </form>
      <span className={styles.copyText}>
        Created with &#9829; by me - &copy; {(new Date()).getFullYear()}
      </span>
    </section>
  );
}

export default Contact;
