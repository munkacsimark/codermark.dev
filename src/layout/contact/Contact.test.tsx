import React from 'react';
import renderer from 'react-test-renderer';
import Contact from './Contact';

it('renders correctly', () => {
  const tree = renderer.create(<Contact />).toJSON();
  expect(tree).toMatchSnapshot();
});
