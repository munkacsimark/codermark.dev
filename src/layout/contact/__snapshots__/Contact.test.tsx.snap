// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="section"
  id="contact"
>
  <h2
    className="title"
  >
    Contact
  </h2>
  <div
    className="text"
  >
    Want to get in touch or talk about a project? Feel free to contact me here.
  </div>
  <form
    action="https://formspree.io/munkacsimark@gmail.com"
    className="form"
    method="POST"
    onSubmit={[Function]}
  >
    <input
      className="input"
      name="name"
      placeholder="What's your name?"
      type="text"
    />
    <input
      className="input"
      name="email"
      placeholder="What's your email?"
      type="email"
    />
    <textarea
      className="textarea"
      placeholder="How can I help you?"
    />
    <button
      className="sendButton"
      type="submit"
    >
      Send message
    </button>
  </form>
  <span
    className="copyText"
  >
    Created with ♥ by me - © 
    2019
  </span>
</section>
`;
